flink-runtime/src/main/java/org/apache/flink/runtime/metrics/dump/MetricQueryService.java

```diff
@@ -70,6 +72,17 @@ public String filterCharacters(String input) {
 	private final Map<Histogram, Tuple2<QueryScopeInfo, String>> histograms = new HashMap<>();
 	private final Map<Meter, Tuple2<QueryScopeInfo, String>> meters = new HashMap<>();
 
+	private long maximumFramesize;
+
+	@Override
+	public void preStart() throws Exception {
+		if (getContext().system().settings().config().hasPath(MAXIMUM_FRAME_SIZE_PATH)) {
```

Comment:
We should rely on `akka.framesize`instead. https://ci.apache.org/projects/flink/flink-docs-master/ops/config.html#akka-framesize

Reply:
`AkkaRpcService` also uses this mode, do you suggest me to refactor it? From the history of git, it is implemented by @tillrohrmann .

Reply:
keep it then, but also copy the comment for the else branch since it is important.